IMAGINGCLOUD-356

s-zubov opened the pull request.

s-zubov committed:
Add on-premise switch to api

s-zubov committed:
Improve code quality

s-zubov committed:
Rename metered to on-premise

s-zubov committed:
Fix tests

s-zubov commented on asposeimagingcloud/api/imaging_api.py:
```diff
@@ -27,27 +27,83 @@
-        if api_version:
-            self.api_client.configuration.api_version = api_version
+
+    def __init__(self, app_key=None, app_sid=None, base_url=None,
```
Python does not support constructor overloads, so I've extended ImagingApi constructor with new optional parameters, making it compatible with previous versions. Also I've added two factory methods `ImagingApi.create_cloud` and `ImagingApi.create_on_premise` with more strict parameters to improve usability.

MaxCrank replied:
@s-zubov would you mind to remove these factory methods (at least for now)? Still, you are free to add a task with a proposition to unify all SDKs by adding such methods and assign it to me - I'll think about it a bit later and if approved, I'll re-assign it to you. Thanks.

s-zubov replied:
@MaxCrank I've removed these methods, now there is only one constructor with multiple optional parameters. `on_premise` parameter is not set explicitly and is determined by input data (`app_sid`, `app_key` and `base_url`).

s-zubov committed:
Add storage name to test_examples

MaxCrank commented on README.md:
```diff
@@ -90,6 +90,20 @@ Please, look at [Examples](EXAMPLES.md) document.
 Each API method has its' async version with *Async* postfix, using *multiprocessing.pool.ThreadPool*.
+Please, look at [Examples](EXAMPLES.md) document
+
+# Aspose Cloud-hosted service VS on-premise deployment (*experimental feature*)
```
@s-zubov Please, move this section to correspond with other SDKs e.g. https://github.com/aspose-imaging-cloud/aspose-imaging-cloud-dotnet/tree/development

s-zubov replied:
@MaxCrank Thank you, I've moved this section.

MaxCrank replied:
@s-zubov It still doesn't look like the one from .NET or other SDKs. Please, inspect the .NET SDK readme https://raw.githubusercontent.com/aspose-imaging-cloud/aspose-imaging-cloud-dotnet/development/README.md and take note on headers

> # Examples
> Please, look at [Examples](EXAMPLES.md) document.
> 
> ### Aspose Cloud-hosted service VS on-premise deployment
> Starting from v19.7, you can choose either to use Aspose Cloud-hosted image processing service (the standard way) or the Docker image from Docker Hub deployed on-premise to serve the requests.
> The details about key differences and deployment process is described on the dedicated Docker Hub page.
> 
> To succeed with your on-premise service usage by the SDK, you need to:
> 1. Use the new API class constructor with base URL required parameter, API version and debug mode optional parameters.
> ```csharp
> var imagingApi = new ImagingApi("yourServiceUrl");
> ```
> 2. Set *storage* or *storageName* parameters for each request where they're present (mandatory!).
> 
> # Tests
> Tests are intended for internal usage only.

MaxCrank commented on README.md:
````diff
@@ -90,6 +90,20 @@ Please, look at [Examples](EXAMPLES.md) document.
+To succeed with your on-premise service usage by the SDK, you need to:
+1. Use the new API class constructor with base URL required parameter, API version and debug mode optional parameters.
+```python
+imagingApi = ImagingApi("yourServiceUrl");
````
@s-zubov this call doesn't seem to correspond with actual constructor which includes _on_premise_ parameter. Please, look at other SDKs which have a single constructor and follow the same approach. Thanks.

s-zubov replied:
@MaxCrank I've refactored constructor, now `ImagingApi("yourServiceUrl");` is a correct call.

s-zubov committed:
Fix extended tests

'False' string is Truly, so it should be parsed and casted to bool to be
Falsy.

s-zubov committed:
Fix tiff tests

s-zubov committed:
Remove ImagingApi factory methods

s-zubov committed:
Fix readme

Commit:
Update README.md
